### ISSUE TITLE
Permission issues

### ISSUE THREAD

**Issue by marcbria** [mbentley/docker-timemachine#14]
First, thanks for the project and to keep it open. 

I cloned locally your repo, then I modify timemachine variables (just the pwd and share_name) and run docker-compose, and the remote server is shown in my Macs, but when I select the server, I get this message:

`"You do not have the necessary read, write and append privileges on the selected network backup disk."`

I double-check permissions (/timemachine folder is owned by 1000 user and group) but I didn't find logs to report it better. The server is a ubuntu 19.04 with your docker and nothing else, and the client is a Mac with 10.6.8.

Any clue about what is happening?

Thanks for your time,
m.

**Comment by mbentley** [comment]
Could you provide the updated Docker compose file (without the password) so I can take a quick look?

**Comment by marcbria** [comment]
Sure. Here it is:

```
version: '3'
services:
  timemachine:
    network_mode: "host"
    environment:
      - CUSTOM_AFP_CONF=false
      - LOG_LEVEL=info
      - MIMIC_MODEL=TimeCapsule6,106
      - TM_USERNAME=timemachine
      - TM_GROUPNAME=timemachine
      - TM_UID=1000
      - TM_GID=1000
      - PASSWORD=xxxxx
      - SET_PERMISSIONS=false
      - SHARE_NAME=TimeMachineSP
      - VOLUME_SIZE_LIMIT=0
    restart: unless-stopped
    ports:
     - "548:548"
     - "636:636"
    volumes:
     - ./volumes/data:/timemachine
     - ./timemachine-netatalk:/var/netatalk
     - ./timemachine-logs:/var/log/supervisor
    ulimits:
      nofile:
        soft: 65536
        hard: 65536
    container_name: timemachine
    image: mbentley/timemachine:latest

```

As said, the only changes are PASSWORD, SHARE_NAME and also , the first volume (I usually place docker data in the same folder as "./volumes/data").

Thanks for your time,
m.

**Comment by marcbria** [comment]
Just in case is useful, permissions in the container's root folder:

```
$ ls -lisah
total 64K
262155 4.0K drwxr-xr-x 6 root   root   4.0K Sep  5 14:02 .
180685 4.0K drwxrwxr-x 5 docker docker 4.0K Sep  5 12:34 ..
262156 4.0K drwxr-xr-x 8 root   root   4.0K Sep  5 10:49 .git
262183 4.0K -rw-r--r-- 1 root   root      6 Sep  5 10:49 .gitignore
262198 4.0K -rw-r--r-- 1 docker docker 1.8K Sep  5 10:49 Dockerfile
262199  12K -rw-r--r-- 1 docker docker  11K Sep  5 10:49 LICENSE
262200 4.0K -rw-r--r-- 1 docker docker 2.7K Sep  5 10:49 README.md
262204 4.0K -rw-r--r-- 1 docker docker  744 Sep  5 10:53 docker-compose.yml
262201 4.0K -rwxr-xr-x 1 docker docker 3.4K Sep  5 10:49 entrypoint.sh
262202 4.0K -rwxr-xr-x 1 docker docker  404 Sep  5 10:49 healthcheck.sh
262203 4.0K -rw-r--r-- 1 docker docker  780 Sep  5 10:49 supervisord.conf
262639 4.0K drwxr-xr-x 2 docker docker 4.0K Sep  5 10:53 timemachine-logs
262640 4.0K drwxr-xr-x 3 docker docker 4.0K Sep  5 10:53 timemachine-netatalk
262154 4.0K drwxr-xr-x 3 docker docker 4.0K Sep  5 10:53 volumes
```

And "volumes/data" is owned by 1000 user and 1000 group.

BTW, I found the logs mapped in ./timemachine-logs. 

I suspect you will like to see the content of netatalk.log:
```
Sep 05 14:23:07.671545 afpd[28] {main.c:149} (info:AFPDaemon): child[7591]: exited 4
Sep 05 14:23:07.671981 afpd[7593] {dsi_tcp.c:241} (info:DSI): AFP/TCP session from 158.109.235.169:62320
Sep 05 14:23:07.672132 afpd[28] {main.c:149} (info:AFPDaemon): child[7592]: exited 4
Sep 05 14:23:24.331972 afpd[28] {main.c:151} (info:AFPDaemon): child[7593]: done
Sep 05 14:23:24.332871 afpd[7612] {dsi_tcp.c:241} (info:DSI): AFP/TCP session from 158.109.235.169:62323
Sep 05 14:23:24.334076 afpd[7612] {uams_dhx2_pam.c:329} (info:UAMS): DHX2 login: timemachine
Sep 05 14:23:24.407547 afpd[7612] {uams_dhx2_pam.c:214} (info:UAMS): PAM DHX2: PAM Success
Sep 05 14:23:24.419380 afpd[7612] {uams_dhx2_pam.c:709} (info:UAMS): DHX2: PAM Auth OK!
Sep 05 14:23:24.419410 afpd[7612] {auth.c:235} (note:AFPDaemon): Login by timemachine (AFP3.3)
Sep 05 14:23:24.481638 cnid_metad[29] {cnid_metad.c:225} (info:CNID): maybe_start_dbd: respawn window ended
Sep 05 14:23:24.896794 afpd[7612] {quota.c:646} (info:AFPDaemon): getquota: special /opt/timemachine fails
Sep 05 14:23:29.317799 afpd[7612] {auth.c:804} (note:AFPDaemon): AFP logout by timemachine
Sep 05 14:23:29.319535 afpd[7612] {afp_dsi.c:108} (note:AFPDaemon): AFP statistics: 1.33 KB read, 1.19 KB written
Sep 05 14:23:29.319564 afpd[7612] {dircache.c:615} (info:AFPDaemon): dircache statistics: entries: 0, lookups: 0, hits: 0, misses: 0, added: 0, removed: 0, expunged: 0, evicted: 0
Sep 05 14:23:29.319619 afpd[7612] {afp_dsi.c:584} (note:AFPDaemon): done
Sep 05 14:23:29.321134 afpd[28] {main.c:151} (info:AFPDaemon): child[7612]: done

```

This "getquota: special /opt/timemachine fails" looks wired.

Cheers,
m.

**Comment by mbentley** [comment]
Looks like there is a problem with the original compose file which is the root of the issue.

Update this:
```
- ./volumes/data:/timemachine
```

To this:
```
- ./volumes/data:/opt/TimeMachineSP
```
and that should work assuming the permissions are correct.

**Comment by marcbria** [comment]
It worked like a charm. Thanks a lot @mbentley.

BTW, it's not something canonical, but I usually set the following structure in my docker projects... somaybe it's also useful for yours:

```
    volumes:
     - ./volumes/data:/opt/timemachine
     - ./volumes/config:/var/netatalk
     - ./volumes/logs:/var/log/supervisor

```

I think it helps newcomers to understand what can be found in each folder and keep permanent volumes together (you know volumes is something to save).

I mean... it works for me, but no ofense at all if you don't find it useful. :-)

Thanks a again for sharing and for your help,
m.

**Comment by ChrisKoh83** [comment]
I had the same issue! But I already had the updated docker-compose.yml and still the permission error.

In my case I used an over username for the login, so I had to change the permissions for this folder:
![Bildschirmfoto 2020-02-08 um 21 04 29](https://user-images.githubusercontent.com/13451391/74091368-f4d00c00-4ab6-11ea-837e-84bd9c62c724.png)

Maybe this is helpful for over users!

**Comment by shred86** [comment]
I had this same issue but it was after I attempted to disable host networking mode and just forward the ports manually. I couldn't get that to work so I switched back to host networking but started receiving this error. Looking at the permissions of the folder that I've mapped to for backups on the host machine, I was seeing:

````
drwxr-sr-x 6 username username 4096 Dec 25 21:00 backup
````

I set the environment variable `SET_PERMISSIONS` to `true` in my docker-compose file, restarted the container and now everything seems to work. The folder permissions are now showing:

````
drwxrwx--- 6 username username 4096 Dec 26 09:12 backup
````

No idea how changing networking modes for the Docker container affected this, but I figured I'd post this here for anyone else that runs into the same issue.
